packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -255,18 +259,33 @@ export function useBlocksStore() {
          */
         const [createdNode] = Editor.parent(editor, editor.selection.anchor.path);
         Transforms.removeNodes(editor, editor.selection);
-        Transforms.insertNodes(editor, {
-          type: 'paragraph',
-          children: createdNode.children,
-        });
+
+        /**
+         * Check if after the current position there is another node
+         */
+        const hasNextNode = editor.children.length - anchorPathInitialPosition[0] > 1;
+
+        /**
+         * Insert the new node at the right position. The next line after the editor selection if present or otherwise at the end of the editor.
+         */
+        Transforms.insertNodes(
+          editor,
+          {
+            type: 'paragraph',
+            children: createdNode.children,
+          },
+          {
+            at: hasNextNode ? [anchorPathInitialPosition[0] + 1] : [editor.children.length],
+          }
+        );
 
         /**
          * The new selection will by default be at the end of the created node.
          * Instead we manually move it to the start of the created node.
          * Use slice(0, -1) to go 1 level higher in the tree,
          * so we go to the start of the node and not the start of the leaf.
          */
-        Transforms.select(editor, editor.start(editor.selection.anchor.path.slice(0, -1)));
+        Transforms.select(editor, editor.start([anchorPathInitialPosition.slice(0, -1)[0] + 1]));
```

Comment:
I am wondering if we directly use anchorPathInitialPosition[0] instead if slicing it, will there be any issue? @remidej

Reply:
I don't think there would be an issue, slice is always a bit confusing so I'd be happy to see it removed